src/engine/polyline_compressor.cpp

```diff
@@ -10,9 +10,19 @@
 namespace osrm::engine::detail // anonymous to keep TU local
 {
 
-std::string encode(int number_to_encode)
```

Comment:
Optimisation 2: we get rid of temporary std::string here via directly encoding numbers to output string.